File: files/CH-163.py
import pandas as pd

path = "CH-163 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=38)
test = pd.read_excel(path, usecols="G:I", skiprows=1, nrows=15)

input['Season'] = pd.to_datetime(input['Year'].astype(str) + " " + input['Month'], format='%Y %b').dt.quarter

result = input.groupby(['Year', 'Season'])['Sale'].sum().reset_index()
result.columns = test.columns = ['Year', 'Season', 'Total Sale']

print(all(result ==test)) # True

File: files/CH-163.R
library(tidyverse)
library(readxl)

path = "files/CH-163 Custom Grouping.xlsx"
input = read_excel(path, range = "B2:D39")
test  = read_excel(path, range = "G2:I17")

result = input %>% 
  unite("ym", Year, Month ,  sep = " ", remove = F) %>%
  mutate(ym = ym(ym),
         Season = quarter(ym)) %>%
  summarise(`Total Sale` = sum(Sale), .by = c(Year, Season))

all.equal(result, test)
# TRUE